### PR TITLE
fix: WAF block based on inputs and IP reputation

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Enable request blocking based on requests that have known bad input
(e.g. the Log4j exploit) or are from IP addresses with a poor
reputation score.

A subsequent PR will be created to enable logging so we can determine
which additional AWS managed rules can be set to block as well.

# Related
* cds-snc/platform-sre-security-support#75

### CODE REVIEW

**Review by jzbahrai — COMMENTED** [review]
its that simple? lgtm but I dont know what I am approving so will wait for someone else.

**Comment by patheard** [comment]
> its that simple? lgtm but I dont know what I am approving so will wait for someone else.

Yup, it's definitely an "easy change, high impact" kind of situation 😅   

If we set all the managed rules to block we'd definitely mess with some legitimate Notify requests because of things like large request body size or missing user agent headers.

File: aws/eks/waf.tf [hunk 1]
```diff
@@ -13,7 +13,7 @@ resource "aws_wafv2_web_acl" "notification-canada-ca" {
     priority = 1
 
     override_action {
-      count {}
+      none {}
```
Comment (jimleroyer) [inline aws/eks/waf.tf:16]: A none overriding action means we would rely on the default allow action, no? I would expect [a `block` block](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/wafv2_web_acl#block) if we want to stop the bots identified by this rule.

File: aws/eks/waf.tf [hunk 2]
```diff
@@ -57,7 +57,7 @@ resource "aws_wafv2_web_acl" "notification-canada-ca" {
     priority = 3
 
     override_action {
-      count {}
+      none {}
```
Comment (jimleroyer) [inline aws/eks/waf.tf:60]: Same here, but maybe I misread the TF/AWS documentation?

File: aws/eks/waf.tf (hunk 1 above)
Comment (patheard) [reply aws/eks/waf.tf:16]: It's a bit confusing - when you use a managed ruleset, you only specify [an `override_action`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/wafv2_web_acl#override-action), which can only have a value of `count` or `none`.  In the `none` case, it means use the managed ruleset's default (`block` in these cases).

Comment (jimleroyer) [reply aws/eks/waf.tf:16]: Definitely confusing, I guess there is no way to know the ruleset's default [other than AWS documentation](https://docs.aws.amazon.com/waf/latest/developerguide/aws-managed-rule-groups-list.html)?

**Review by jimleroyer — APPROVED** [review]

**Pull request merged**